### PR TITLE
crypto: use secp256k1 instead of k256

### PR DESCRIPTION
**Pull request by joyqvq**

### CODE REVIEW

**Comment by huitseeker** [comment]
@joyqvq could you run the micro-benchmark and report the result in the PR message?
Criterion has a baseline function and auxiliary crates like criterion-benchcmp which may be useful for comparisons.

**Comment by joyqvq** [comment]
```
===========
k256 impl
===========
Secp256k1 signing       time:   [62.989 µs 63.180 µs 63.414 µs]
Found 6 outliers among 100 measurements (6.00%)
  4 (4.00%) high mild
  2 (2.00%) high severe

Secp256k1 signature verification
                        time:   [75.266 µs 75.386 µs 75.554 µs]
Found 17 outliers among 100 measurements (17.00%)
  2 (2.00%) high mild
  15 (15.00%) high severe

======
rust binding to FFI
======
Secp256k1 signing       time:   [43.666 µs 43.728 µs 43.811 µs]
                        change: [-31.026% -30.868% -30.718%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 10 outliers among 100 measurements (10.00%)
  3 (3.00%) high mild
  7 (7.00%) high severe


Secp256k1 signature verification
                        time:   [50.841 µs 50.944 µs 51.083 µs]
                        change: [-33.033% -32.793% -32.568%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 10 outliers among 100 measurements (10.00%)
  3 (3.00%) high mild
  7 (7.00%) high severe
```

saving ~30% on signing and ~33% on verification

**Review by huitseeker — APPROVED** [review]
Nice work! Thanks @joyqvq !

File: crypto/src/secp256k1.rs [hunk 1]
```diff
@@ -288,8 +289,9 @@ impl KeyPair for Secp256k1KeyPair {
     }
 
     fn generate<R: rand::CryptoRng + rand::RngCore>(_rng: &mut R) -> Self {
-        let privkey = k256::ecdsa::SigningKey::random(&mut OsRng);
-        let pubkey = k256::ecdsa::VerifyingKey::from(&privkey);
+        let secp = Secp256k1::new();
+        // TODO: use param rng instead of generate a fresh OsRng when rand is upgraded to match. https://github.com/MystenLabs/narwhal/issues/544
```
Comment (kchalkias) [inline crypto/src/secp256k1.rs:293]: +1

**Review by kchalkias — APPROVED** [review]
Awesome! towards faster crypto!

**Pull request merged**